pkg/controller/scyllacluster/resource/resource_test.go

```diff
@@ -314,3 +319,419 @@ func TestMemberService(t *testing.T) {
+								},
+								Ports: []corev1.ContainerPort{
+									{
+										Name:          "agent-rest-api",
+										ContainerPort: 10001,
+									},
+								},
+								VolumeMounts: []corev1.VolumeMount{
+									{
+										Name:      naming.PVCTemplateName,
+										MountPath: naming.DataDir,
+									},
+									{
+										Name:      scyllaAgentConfigVolumeName,
+										MountPath: path.Join(naming.ScyllaAgentConfigDirName, naming.ScyllaAgentConfigFileName),
+										SubPath:   naming.ScyllaAgentConfigFileName,
+										ReadOnly:  true,
+									},
+									{
+										Name:      scyllaAgentAuthTokenVolumeName,
+										MountPath: path.Join(naming.ScyllaAgentConfigDirName, naming.ScyllaAgentAuthTokenFileName),
+										SubPath:   naming.ScyllaAgentAuthTokenFileName,
+										ReadOnly:  true,
+									},
+								},
+								Resources: basicRack.Resources,
+							},
+						},
+						DNSPolicy:          "ClusterFirstWithHostNet",
+						ServiceAccountName: "basic-member",
+						Affinity:           &corev1.Affinity{},
+					},
+				},
+				VolumeClaimTemplates: []corev1.PersistentVolumeClaim{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:   naming.PVCTemplateName,
+							Labels: basicSTSLabels(),
+						},
+						Spec: corev1.PersistentVolumeClaimSpec{
+							AccessModes:      []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+							StorageClassName: basicRack.Storage.StorageClassName,
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									corev1.ResourceStorage: resource.MustParse(basicRack.Storage.Capacity),
+								},
+							},
+						},
+					},
+				},
+				ServiceName:         naming.HeadlessServiceNameForCluster(basicSC()),
+				PodManagementPolicy: appsv1.OrderedReadyPodManagement,
+				UpdateStrategy: appsv1.StatefulSetUpdateStrategy{
+					Type: appsv1.RollingUpdateStatefulSetStrategyType,
+					RollingUpdate: &appsv1.RollingUpdateStatefulSetStrategy{
+						Partition: pointer.Int32Ptr(0),
+					},
+				},
+			},
+		}
+	}
+
+	var existingSts *appsv1.StatefulSet = nil
+	sidecarImage := ""
+
+	tt := []struct {
+		name          string
+		rack          scyllav1.RackSpec
+		scyllaCluster *scyllav1.ScyllaCluster
+		existingSts   *appsv1.StatefulSet
+		sidecarImage  string
+		expectedSts   *appsv1.StatefulSet
+	}{
+		{
+			name:          "basic STS",
+			rack:          basicRack,
+			scyllaCluster: basicSC(),
+			existingSts:   existingSts,
+			sidecarImage:  sidecarImage,
+			expectedSts:   newSts(),
+		},
+		{
+			name: "non-nil ImagePullSecrets",
+			rack: basicRack,
+			scyllaCluster: func() *scyllav1.ScyllaCluster {
+				sc := basicSC()
+				sc.Spec.ImagePullSecrets = basicImagePullSecrets()
+				return sc
+			}(),
+			existingSts:  existingSts,
+			sidecarImage: sidecarImage,
+			expectedSts: func() *appsv1.StatefulSet {
+				sts := newSts()
+				sts.Spec.Template.Spec.ImagePullSecrets = basicImagePullSecrets()
```

Comment:
would it be better to inline the expected value for the test? if someone accidentally makes the function `basicImagePullSecrets()` to return empty array, the test will still work, although it claims `"non-nil ImagePullSecrets`

Reply:
done